### PR TITLE
EVG-12866: Add performance-data.send command to evergreen

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-12866

- vendor juniper and poplar, revendor timber to enable using multiple grpc protobuf files from cedar.
- add `perf.send` command to send performance data directly to cedar via evg.

### CODE REVIEW

File: command/perf_send.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/rest/client"
+	"github.com/evergreen-ci/poplar"
+	"github.com/evergreen-ci/poplar/rpc"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+)
+
+type perfSend struct {
+	// AwsKey and AwsSecret are the user's credentials for authenticating
```
Comment (Kimchelly) [inline command/perf_send.go:16]: AWSKey and AWSSecret

File: command/perf_send.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	// without a bucket specified.
+	Bucket string `mapstructure:"bucket" plugin:"expand"`
+
+	// Prefix specifies the global prefix to use within the s3 bucket for
+	// any artifacts without a prefix specified.
+	Prefix string `mapstructure:"region" plugin:"expand"`
+
+	// File is the file containing either the json or yaml representation
+	// of the performance report tests.
+	File string `mapstructure:"file" plugin:"expand"`
```
Comment (Kimchelly) [inline command/perf_send.go:36]: Does it have to be all in one file? Do they have the option of uploading multiple files, or would this never happen?

File: command/perf_send.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+type perfSend struct {
+	// AwsKey and AwsSecret are the user's credentials for authenticating
+	// interactions with s3. These are required if any of the tests have
+	// artifacts.
+	AWSKey    string `mapstructure:"aws_key" plugin:"expand"`
+	AWSSecret string `mapstructure:"aws_secret" plugin:"expand"`
+
+	// Region is the s3 region where the global bucket is located. It
+	// defaults to "us-east-1".
```
Comment (Kimchelly) [inline command/perf_send.go:23]: Where does it actually set the default region? Is it within Poplar/Cedar?

File: glide.yaml [hunk 4]
```diff
@@ -57,6 +57,7 @@ import:
   - package: github.com/evergreen-ci/pail
   - package: gonum.org/v1/gonum
   - package: github.com/evergreen-ci/timber
+  - package: github.com/evergreen-ci/poplar
```
Comment (Kimchelly) [inline glide.yaml:61]: Also include juniper.

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: command/perf_send.go (hunk 2 above)
Comment (julianedwards) [reply command/perf_send.go:36]: It is all in one file, not possible to have multiple files.

File: command/perf_send.go (hunk 3 above)
Comment (julianedwards) [reply command/perf_send.go:23]: Within poplar.

**Review by Kimchelly — APPROVED** [review]

**Pull request merged**